src/compiler/crystal/semantic/ast.cr

```diff
@@ -4,7 +4,6 @@ module Crystal
   def self.check_type_allowed_in_generics(node, type, msg)
     return if type.allowed_in_generics?
 
-    type = type.union_types.find { |t| !t.allowed_in_generics? } if type.is_a?(UnionType)
```

Comment:
I would rather check through commit history about this change

Reply:
I did. And from this commit bf284d99fef83 i didn't get why we need to do this useless assignment.